### PR TITLE
enhance perf_hv_gpci

### PR DESCRIPTION
**Pull request by disgoel**
- enhanced test to fail with not supported event count 
- added negative testcase to run hv_gpci events with non-root user which should not be able to read counter data 
- removing hardcoded value for partition_id and getting the value from lparcfg file 
- removing per core, per socket option as its not supported with hv_gpci, it collects data system wide

[root@]# avocado run perf_hv_gpci.py
JOB ID     : 47c92b52a1be61ef9d04996ddb727a954d56e3c0
JOB LOG    : /home/abc/avocado-fvt-wrapper/results/job-2023-04-05T06.52-47c92b5/job.log
 (1/1) perf_hv_gpci.py:perf_hv_gpci.test_gpci_events: STARTED
 (1/1) perf_hv_gpci.py:perf_hv_gpci.test_gpci_events: PASS (84.04 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/abc/avocado-fvt-wrapper/results/job-2023-04-05T06.52-47c92b5/results.html
JOB TIME   : 111.85 s
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11158630/job.log)

test fails with not supported events:
[root@]# avocado run perf_hv_gpci.py
JOB ID     : 789be9cc05f42c766850ea0d258ea82e2165fec3
JOB LOG    : /home/disha/avocado-fvt-wrapper/results/job-2023-04-05T17.51-789be9c/job.log
 (1/1) perf_hv_gpci.py:perf_hv_gpci.test_gpci_events: STARTED
 (1/1) perf_hv_gpci.py:perf_hv_gpci.test_gpci_events: FAIL: perf hv_gpci: some of the events failed,refer to log (16.53 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/disha/avocado-fvt-wrapper/results/job-2023-04-05T17.51-789be9c/results.html
JOB TIME   : 51.43 s
[debug.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11158721/debug.log)

### CODE REVIEW

File: perf/perf_hv_gpci.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def setUp(self):
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
 
+        # create temporary user
+        process.run('useradd test', sudo=True, ignore_status=True)
```
Comment (PraveenPenguin) [inline perf/perf_hv_gpci.py:62]: what about if test user creation failed due to some issue?

File: perf/perf_hv_gpci.py [hunk 2]
```diff
@@ -85,14 +88,24 @@ def error_check(self):
...
-        if output.exit_status != 0:
+        result = process.run(cmd, shell=True, ignore_status=True)
+        output = result.stdout.decode() + result.stderr.decode()
+        if (result.exit_status != 0) or ("not supported" in output):
             self.fail_cmd.append(cmd)
 
+        # test hv_gpci events with normal user
+        result = process.run("su - test -c '%s'" %
+                             cmd, shell=True, ignore_status=True)
+        output = result.stdout.decode() + result.stderr.decode()
```
Comment (PraveenPenguin) [inline perf/perf_hv_gpci.py:103]: before running this command we need to check if test user present

File: perf/perf_hv_gpci.py [hunk 3]
```diff
@@ -124,5 +137,6 @@ def test_gpci_events(self):
             self.gpci_events(event)
 
     def tearDown(self):
+        process.run('userdel -f test', sudo=True, ignore_status=True)
```
Comment (PraveenPenguin) [inline perf/perf_hv_gpci.py:140]: please have if user present

Comment (disgoel) [reply perf/perf_hv_gpci.py]: @PraveenPenguin have addressed all review comments and updated, please check.

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**